File: Data Structures & Algorithms/same-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if (!p && !q) return true;

        queue<pair<TreeNode*, TreeNode*>> queue;
        queue.push({p,q});

        while (!queue.empty()) {
            pair<TreeNode*, TreeNode*> pie = queue.front();
            queue.pop();

            if (!pie.first && !pie.second) continue;
            if (!pie.first || !pie.second || pie.first -> val != pie.second -> val) return false;

            queue.push({pie.first -> left, pie.second -> left});
            queue.push({pie.first -> right, pie.second -> right});

        }

        return true;

        
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


# good node if node is greater than prior node 
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = 0

        def loop(prevmax:int, root: TreeNode):
            nonlocal res
            if prevmax <= root.val:
                res += 1


            if root.left:
                loop(max(prevmax, root.val), root.left)
            if root.right:
                loop(max(prevmax, root.val), root.right)
            
        loop(root.val, root)
        
        return res
        
            
